.yarn/patches/@metamask-assets-controllers-npm-38.2.0-40af2afaa7.patch

```diff
@@ -0,0 +1,36 @@
+diff --git a/dist/assetsUtil.cjs b/dist/assetsUtil.cjs
+index e90a1b6767bc8ac54b7a4d580035cf5db6861dca..2662df7869e9079f9379c4dd1ded40b5af4e71f2 100644
+--- a/dist/assetsUtil.cjs
++++ b/dist/assetsUtil.cjs
+@@ -3,6 +3,7 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
+     return (mod && mod.__esModule) ? mod : { "default": mod };
+ };
+ Object.defineProperty(exports, "__esModule", { value: true });
++function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { newObj[key] = obj[key]; } } } newObj.default = obj; return newObj; } }
+ exports.fetchTokenContractExchangeRates = exports.reduceInBatchesSerially = exports.divideIntoBatches = exports.ethersBigNumberToBN = exports.addUrlProtocolPrefix = exports.getFormattedIpfsUrl = exports.getIpfsCIDv1AndPath = exports.removeIpfsProtocolPrefix = exports.isTokenListSupportedForNetwork = exports.isTokenDetectionSupportedForNetwork = exports.SupportedTokenDetectionNetworks = exports.formatIconUrlWithProxy = exports.formatAggregatorNames = exports.hasNewCollectionFields = exports.compareNftMetadata = exports.TOKEN_PRICES_BATCH_SIZE = void 0;
+ const controller_utils_1 = require("@metamask/controller-utils");
+ const utils_1 = require("@metamask/utils");
+@@ -221,7 +222,7 @@ async function getIpfsCIDv1AndPath(ipfsUrl) {
+     const index = url.indexOf('/');
+     const cid = index !== -1 ? url.substring(0, index) : url;
+     const path = index !== -1 ? url.substring(index) : undefined;
+-    const { CID } = await import("multiformats");
++    const { CID } = await Promise.resolve().then(() => _interopRequireWildcard(require("multiformats")));
```

Comment:
Going back to the change itself:

The await on an empty promise before running synchronous require seems unnecessary. Does anything depend on this step being asynchronous? Doesn't seem so.
```suggestion
+    const { CID } = _interopRequireWildcard(require("multiformats"));
```

Reply:
I think in [v37](https://github.com/MetaMask/core/blob/main/packages/assets-controllers/CHANGELOG.md#3700) this change was introduced:
BREAKING: getIpfsCIDv1AndPath, getFormattedIpfsUrl are now async functions (https://github.com/MetaMask/core/pull/3645)

This patch update matches the generated build after the V37 update;

Reply:
The `await` is necessary in the CJS source code since we're forced to use a dynamic import. But in the build output that we're patching we definitely don't need to both `then` and `await` the synchronous _interopRequireWildcard` call. 

@naugtur's suggestion looks good to me, although I'd feel better with another manual check on whether CID is sourced at runtime.

Reply:
Thank you @MajorLift and @naugtur  🙏  i have removed the empty promise; could you please approve again and we merge this 🙏